frame/assets/src/types.rs

```diff
@@ -185,3 +187,39 @@ impl From<TransferFlags> for DebitFlags {
 		}
 	}
 }
+
+/// Possible errors when converting between external and asset balances.
+#[derive(Eq, PartialEq, Copy, Clone, RuntimeDebug, Encode, Decode)]
+pub enum ConversionError {
+	/// The asset is not present in storage.
+	AssetMissing,
+	/// The asset is not sufficient and thus does not have a reliable `min_balance` so it cannot be converted.
+	AssetNotSufficient,
+}
+
+/// Converts a balance value into an asset balance based on the ratio between the existential
+/// deposit and the minimum asset balance.
+pub struct BalanceToAssetBalance<T, Balance, ED>(PhantomData<(T, Balance, ED)>);
```

Comment:
```suggestion
pub struct BalanceToAssetBalance<T, Balances: Currency>(PhantomData<(T, Balance, ED)>);
```

Instead of `Balance` and `ED` separately you can accept something that is `Currency` and then use `Currency::minimum_balance()` and `Currency::Balance`.